WorkloadManagementSystem/Service/JobManagerHandler.py

```diff
@@ -240,6 +246,22 @@ def __deleteJob( self, jobID ):
     if not result['OK']:
       gLogger.warn( 'Failed to delete job from the TaskQueue' )
 
+    # if it was the last job for the pilot, clear PilotsLogging about it
+    result = gPilotAgentsDB.getPilotsForJobID( jobID )
+    if not result['OK']:
+      return result
+    for pilot in result['Value']:
+        res = gPilotAgentsDB.getJobsForPilot( pilot[ 'PilotID' ] )
+        if not res['OK']:
+            return res
+        if not res['Value']: # if list of jobs for pilot is empty, delete pilot and pilotslogging
+            ret = gPilotAgentsDB.deletePilot( pilot[ 'PilotID' ] )
```

Comment:
Well, this is deleting pilots when jobs are deleted, so it won't affect the deletion of pilots that finished with no jobs.